process_report/process_report.py

```diff
@@ -1,9 +1,108 @@
+    credit_code = "0002"
+    credit_amount = 1000
+
+    old_pi_list = set()
+    try:
+        with open(os.getenv("C0002_OLD_PI")) as f:
+            for pi in f: old_pi_list.add(pi.strip())
+
+    except Exception:
+        print("Old PI file does not exist, or the C0002_OLD_PI env var is not set")
+        sys.exit(1)
+
+    pi_list = dataframe[MANAGER_PI_FIELD].unique()
+
+    for pi in pi_list:
+        if pi != pi: continue   # NaN check
+        if pi in old_pi_list: continue  # Is the PI an old PI?
+
+        pi_projects = dataframe[dataframe[MANAGER_PI_FIELD] == pi]
+        rem_credit = credit_amount
+        for i, row in pi_projects.iterrows():
+            project_cost = row[COST_FIELD]
+            if project_cost >= rem_credit:
+                dataframe.at[i, CREDIT_FIELD] =  rem_credit
+                dataframe.at[i, CREDIT_CODE_FIELD] = credit_code
+                dataframe.at[i, BALANCE_FIELD] = row[COST_FIELD] - rem_credit
+                break
+            else:
+                dataframe.at[i, CREDIT_FIELD] =  project_cost
+                dataframe.at[i, CREDIT_CODE_FIELD] = credit_code
+                dataframe.at[i, BALANCE_FIELD] = 0
+                rem_credit -= project_cost
+    
+    return dataframe
+
+
+applied_credits = [apply_credits_0002]
+
+
+def get_institution_from_pi(pi_uname):
+    
+    institute_map = {
+        "northeastern.edu"      : "Northeastern University",
+        "bu.edu"                : "Boston University",
+        "bentley.edu"           : "Bentley",
+        "uri.edu"               : "University of Rhode Island",
+        "redhat.com"            : "Red Hat",
+        "childrens.harvard.edu" : "Boston Childrens Hospital",
+        "mclean.harvard.edu"    : "McLean Hospital",
+        "meei.harvard.edu"      : "Massachusetts Eye & Ear",
+        "dfci.harvard.edu"      : "Dana-Farber Cancer Institute",
+        "bwh.harvard.edu"       : "Brigham and Women's Hospital",
+        "bidmc.harvard.edu"     : "Beth Israel Deaconess Medical Center",
+        "harvard.edu"           : "Harvard University",
+        "wpi.edu"               : "Worcester Polytechnic Institute",
+        "mit.edu"               : "Massachusetts Institute of Technology",
+        "umass.edu"             : "University of Massachusetts Amherst",
+        "uml.edu"               : "University of Massachusetts Lowell",
+        "codeforboston.org"     : "Code For Boston",
+        "mmsh"                  : "Harvard University",
+        "gstuart"               : "University of Massachusetts Amherst",
+        "rudolph"               : "Boston Childrens Hospital",
+        "robbaron"              : "Boston University",
+        "kmdalton"              : "Harvard University",
+        "mzink"                 : "University of Massachusetts Amherst",
+        "yale.edu"              : "Yale University",
+        "francesco.pontiggia"   : "Harvard University",
```

Comment:
This and a bunch of others above it are not institute domains but rather people's name. Why do we have this here?

Reply:
@naved001 Some PIs have set their usernames to be...usernames, not they institute-specific email. As far as I know, the only way to know what institute they belong to is to hardcode their usernames in this mapping.